### PR TITLE
93 scs rainfall curve back end

### PR DESCRIPTION
**Pull request by GundulaW**

### CODE REVIEW

**Comment by GundulaW** [comment]
Added rainfall shapes including SCS rainfall curves in the back-end. 

The following tests are still failing but Tim has fixed these in the hurricane branch:

- test_object_mode\test_events\test_download_meteo
- test_object_mode\test_integrator\test_preprocess_rainfall_timeseries --> was working when Tim checked out my branch but we used the same hydromt versions

**Comment by GundulaW** [comment]
I also had to made some changes to event and hazard.py related to event_sets

File: flood_adapt/object_model/hazard/event/event.py [hunk 1]
```diff
@@ -49,6 +51,30 @@ def get_mode(filepath: Path) -> str:
         mode = event_data["mode"]
         return mode
 
+    @staticmethod
+    def timeseries_shape(
+        shape_type: str, duration: float, peak: float, **kwargs
+    ) -> np.ndarray:
```
Comment (panosatha) [inline flood_adapt/object_model/hazard/event/event.py:57]: could you write some doc strings in the functions in this class?

File: flood_adapt/object_model/hazard/event/synthetic.py [hunk 2]
```diff
@@ -93,7 +93,7 @@ def add_tide_and_surge_ts(self):
             time_shift = (
                 self.attrs.time.duration_before_t0 + self.attrs.surge.shape_peak_time
             ) * 3600
-            surge = self.timeseries_shape(
+            surge = super().timeseries_shape(
```
Comment (panosatha) [inline flood_adapt/object_model/hazard/event/synthetic.py:96]: why do you need to use super() here? self wouldn't work?

File: tests/test_database/charleston/static/site/scs_rainfall.csv
Comment (panosatha) [inline tests/test_database/charleston/static/site/scs_rainfall.csv:1]: I would put this in a seperate "rainfall_curves" folder in the static data folder

**Review by panosatha — CHANGES_REQUESTED** [review]
Great work! I made some minor comments that you can have a look before merging.

File: .gitignore [hunk 3]
```diff
@@ -170,4 +170,5 @@ tests/test_database/charleston/output/
 tests/test_database/charleston/static/templates/overland/
 tests/test_database/charleston/input/measures/raise_property_aggregation_area/raise_property_aggregation_area_new.toml
 tests/system/sfincs/
-database
\ No newline at end of file
+database
+tests/test_database/charleston/input/events/
```
[the remark this answers is not in the document]
Comment (GundulaW) [reply .gitignore:174]: removed this

**Pull request merged**